docs(libevm/tooling/release): add readme (#170)
ava-labs/go-ethereum: libevm/tooling/release/README.md

# Release

> In the following, we create a release candidate tag `v1.13.14-0.2.0.rc.4`.

Reviewer: ```suggestion
In the following, we create a release candidate version `v1.13.14-0.2.0.rc.4`.
```

> 1. Create two branches, usually from the `main` branch:

Reviewer: ```suggestion
1. Create two branches, usually from the tip of the `main` branch, but the base commit MAY be from anywhere on `main`:
```

    ```bash
    git fetch origin main:main
    git checkout main

>     git checkout -b release/v1.13.14-0.2.0.rc.4

Reviewer: Lets add a step before that sets an environment variable and then use it in the rest. That helps someone copy-paste with fewer modifications.

    git push -u origin release/v1.13.14-0.2.0.rc.4
    git checkout -b myusername/release/v1.13.14-0.2.0.rc.4

>     git push -u origin myusername/release/v1.13.14-0.2.0.rc.4

Reviewer: This step is premature and unnecessary at this point.

    ```

    The `myusername/release/v1.13.14-0.2.0.rc.4` branch will be used to add "release modifications" and will target the branch `release/v1.13.14-0.2.0.rc.4` in a pull request.

> 1. Run script `./cherrypick.sh` which cherry picks all Geth commits listed in [cherrypicks](cherrypicks)
>     - you may have to resolve conflicts

Reviewer: ```suggestion
1. Run the script `./cherrypick.sh` which cherry picks all Geth commits listed in [cherrypicks](cherrypicks). You may have to resolve conflicts.
```

Author: And also add that you should resolve compilation errors and test errors by amending the cherry pick commit.
I had the case, a cherry pick commit would produce a compilation error, we don't want to add "fix commits" after the cherry-picked commits really ideally I suppose.

> 1. Modify [params/version.libevm.go](/params/version.libevm.go)

Reviewer: It's useful to have this as the first step but NOT have anything committed. That way the version string is automatically generated for naming the release branch.

Author: > That way the version string is automatically generated for naming the release branch.

Through a test right?
I think it's best to keep it as it is, and have the CI check on the version string humanly typed, to avoid any bad copy-paste surprises.

>     - Change the `LibEVMVersion` to the correct final version name

Reviewer: ```suggestion
    - Change the `LibEVMVersion` to the correct final version string (`version.libevm_test.go` will ensure that this is correctly formatted)
```

And also make it the last point.

Author: Bullet points are meant to have no order unlike numbered points, but sure, done 👍

    - Change the `LibEVMReleaseType` to the correct release type, for example `ReleaseCandidate`
    - If planning a release candidate: set `libEVMReleaseCandidate` to the correct number; in this case `4`
    - If needed: change the `LibEVMVersionMajor`, `LibEVMVersionMinor` and `LibEVMVersionPatch` numbers
1. Commit your modifications to [params/version.libevm.go](/params/version.libevm.go) with a commit title `chore: release v1.13.14-0.2.0.rc.4`.

> 1. Push your modified branch to the remote `git push`

Reviewer: ```suggestion
1. Push your modified branch to the remote `git push -u origin myusername/release/v1.13.14-0.2.0.rc.4`
```

Since we're removing it from earlier.

> 1. Open a pull request from your modified branch `myusername/release/v1.13.14-0.2.0.rc.4` and targeting `release/v1.13.14-0.2.0.rc.4`, for example with `https://github.com/ava-labs/libevm/compare/release/v1.13.14-0.2.0.rc.4...myusername/release/v1.13.14-0.2.0.rc.4?expand=1`. Set the tile to "chore: release `v1.13.14-0.2.0.rc.4`"

Reviewer: ```suggestion
1. Open a pull request from your modified branch `myusername/release/v1.13.14-0.2.0.rc.4` and targeting `release/v1.13.14-0.2.0.rc.4`, for example with `https://github.com/ava-labs/libevm/compare/release/v1.13.14-0.2.0.rc.4...myusername/release/v1.13.14-0.2.0.rc.4?expand=1`.
    1. Set the tile to "chore: release `v1.13.14-0.2.0.rc.4`"
```

Author: This looks a bit strange to have a single sub-point `a.` don't you think?

Author: It's now sub-numbers for other reasons

# Release

In the following, we create a release candidate version `v1.13.14-0.2.0.rc.4` and your Github username is `myusername`. We set environment variables accordingly:

```bash
VERSION=v1.13.14-0.2.0.rc.4

> USERNAME=myusername

Author: Change to GH_USER or it conflicts with OSX `$USERNAME`
